qsiprep/tests/test_utils_grouping.py

```diff
@@ -117,6 +185,44 @@ def test_get_entity_groups_without_multipartid(tmpdir):
     check_expected(entity_groups, expected)
 
 
+def test_get_fieldmaps(tmp_path_factory):
+    """Test the get_fieldmaps function."""
+    base_dir = tmp_path_factory.mktemp('test_get_fieldmaps')
+
+    # Check that relative paths are correctly handled
+    bids_dir = base_dir / 'dset_fmap_intendedfor_relpath'
+    generate_bids_skeleton(str(bids_dir), dset_fmap_intendedfor_relpath)
+    layout = BIDSLayout(str(bids_dir))
+    dwi_file = layout.get(suffix='dwi', extension='nii.gz', return_type='filename')[0]
+    fieldmaps = layout.get_fieldmap(dwi_file, return_list=True)
+    assert len(fieldmaps) == 1
+    assert fieldmaps[0]['suffix'] == 'epi'
+    assert layout.get_file(fieldmaps[0]['epi']).get_metadata()['IntendedFor'] == [
+        'dwi/sub-01_dir-AP_dwi.nii.gz'
+    ]
+
+    # Check that BIDS URI paths are correctly handled
+    bids_dir = base_dir / 'dset_fmap_intendedfor_bidsuri'
+    generate_bids_skeleton(str(bids_dir), dset_fmap_intendedfor_bidsuri)
+    layout = BIDSLayout(str(bids_dir))
+    dwi_file = layout.get(suffix='dwi', extension='nii.gz', return_type='filename')[0]
+    fieldmaps = layout.get_fieldmap(dwi_file, return_list=True)
```

Comment:
`BIDSLayout.get_fieldmap` doesn't work on BIDS URIs, so we need to either work around it like SDCFlows does _or_ contribute to PyBIDS to make it work.